pkg/trace/stats/concentrator_test.go

```diff
@@ -584,6 +598,48 @@ func TestConcentratorStatsCounts(t *testing.T) {
 	}
 }
 
+// TestRootTag tests that an aggregation will be slit up by the IsTraceRoot aggKey
+func TestRootTag(t *testing.T) {
+	now := time.Now()
+	spans := []*pb.Span{
+		testSpan(now, 1, 0, 40, 10, "A1", "resource1", 0, nil),
+		testSpan(now, 1, 0, 40, 10, "A1", "resource1", 0, nil),
+		testSpan(now, 1, 100, 30, 10, "A1", "resource1", 0, nil),
```

Comment:
This is a trace where we have 3 identical looking spans, two of which are root spans and one span is orphaned (its parent is `100` but there's no span here with that ID). It might be odd to have two root spans in a trace.

Let's set something up that is a little more of what we'd expect:
- First span as is
- Second span has span ID of 2 and parent ID of 1 (so it's a child of the first span)
- Third span is a child of the second span and has a span ID of 3; let's also pass a `map[string]string` as below in order to mark it as a client span
- Fourth span can be an orphaned case like the third span today

Here's the map to use for the third span:
```
map[string]string{"span.kind": "client"}
```

I would expect stats for only these spans:
- The first span because it's a root, top-level span
- The third span because it's a client span
- The fourth span because it gets marked as a top-level (part of the logic for orphaned spans, see here: https://github.com/DataDog/datadog-agent/blob/cec4dfe545f33e57d503c0c4ee685878794ab310/pkg/trace/traceutil/trace.go#L96)

Only the stats for the first and fourth spans will have `IsTraceRoot: true` and the stats for the third span will have `IsTraceRoot: false`.

Reply:
You may find it preferable to make your own spans instead of relying on `testSpan`. They should be simple enough structs.

Reply:
Reworked test case